src/utils/plotSpecs/generateViolinSpec.js

```diff
@@ -477,19 +477,20 @@ const generateData = (
 
   cellSetsIds.forEach((cellSetId) => {
     const currentCellIds = Array.from(properties[cellSetId].cellIds);
-
     currentCellIds
       .filter(shouldBeDisplayed)
       .forEach((cellId) => {
-        const cell = {
-          group: cellSetId,
-          y: selectedExpression[cellId],
-        };
+        if (selectedExpression[cellId]) {
```

Comment:
In some cases the cellIds included all the unfiltered cells. This was causing the y axis points to be mostly undefined. Not sure the unfiltered cells  are still stored in redux, but this if avoids ploting y undefined values.